hpx/runtime/threads/thread_data_fwd.hpp

```diff
@@ -167,5 +171,18 @@ namespace hpx { namespace threads
 #endif
 }}
 
+namespace std {
+    template <>
+    struct hash<::hpx::threads::thread_id_type>
+    {
+        std::size_t operator()(::hpx::threads::thread_id_type const& v) const
+            noexcept
+        {
+            std::hash<const ::hpx::threads::thread_data*> hasher_;
```

Comment:
Not that I want to kick off a religious discussion here... But doesn't all of HPX use `east const` notations?

Reply:
No need for religion :) The const is where it always was and I'll happily move it.

Reply:
We've had a `west const` in our code base? Unheard of! ;-)